docs/topics/settings.txt
.. _topics-settings:

==========================
Full list of settings
==========================

.. module:: djpcms.settings

A Django settings file doesn't have to define any djpcms settings if it doesn't need
to. Each setting has a sensible default value. These defaults live in the
module :file:`djpcms/settings.py`.

Here's a full list of all available settings, in alphabetical order, and their
default values.


.. setting:: DJPCMS_MARKUP_MODULE

DJPCMS_MARKUP_MODULE
-------------------------

default: ``'djpcms.utils.markups'``

Python dotted path to the markup module. This module exposes four functions:


.. setting:: DJPCMS_PLUGINS

DJPCMS_PLUGINS
----------------------

Default: ``['djpcms.plugins.*]``

A list of python dotted paths to where plugins are defined.



.. setting:: DJPCMS_PLUGIN_BASE_URL

DJPCMS_PLUGIN_BASE_URL
------------------------

Default: ``'/plugin/'``

String used as base url when constructing plugin urls. Make sure to use a trailing slash, and to have
this be different from any other url you may use in your application.


See: :ref:`topics-plugins-url`



.. setting:: DEFAULT_VIEW_MODULE

DEFAULT_VIEW_MODULE
----------------------

Default: ``djpcms.views.pageview.pageview``

The python dotted path to the default python class used to handle view requests.



.. setting:: HTML_CLASSES

HTML_CLASSES
----------------

Default: ``djpcms.ajaxhtml.ajaxhtml()``

Instance of djpcms.ajaxhtml.ajaxhtml. It contains a dictionary of css classes which may define particular
javascript decorators or simple css styling.

See: :ref:`topics-javascript-index`



.. setting:: SERVE_STATIC_FILES

SERVE_STATIC_FILES
----------------------

Default: ``False``

Whether static files are served by the server running the Django site (and therefore included in the urls) or not.

It should be set to True only during development.



.. setting:: CONTENT_INLINE_EDITING

CONTENT_INLINE_EDITING
-------------------------

Dictionary of information for inline content editing, by default it is set for no editing::

	CONTENT_INLINE_EDITING = {
				'available': False,
				'preurl': 'edit',
				'permission': None,
				'pagecontent': '/content/',
				'width': 600,
				'height': 400
				}

The ``pagecontent`` key define the base url for editing page contents. It should contains trailing slashes.
A :ref:`topics-models-page` object with url given by ``pagecontent`` needs to be created if ``available`` is set to True.
Otherwise an error will occur. 



.. setting:: ENABLE_BREADCRUMBS

ENABLE_BREADCRUMBS
-------------------------
Default: ``1``

If set to 0, no bread crumbs support, otherwise it indicates from which level to start generating
bread-crumbs (usually 1 or 2).


.. settings:: GRID960_DEFAULT_FIXED

GRID960_DEFAULT_FIXED
--------------------------
Default: ``True``

If set to True (the default), the grid 960 layout is fixed at 960px, otherwise is floating to the whole page.
See `960 grid system <http://960.gs/>`_ and `fluid 960 grid system <http://www.designinfluences.com/fluid960gs/>`_ 



.. _html-classes:

HTML_CLASSES
-------------------------

This object contains names of classes used for the interaction between Django and jQuery



.. _google_analytics_id:

GOOGLE_ANALYTICS_ID
-------------------------
Default: ``None``

ID for Google Analytics.



CHANGELOG.rst

0.4 (2009-12-24)
=====================
first official Alpha release.

0.5
=====================

 * Bug fixes
 * Added "splitregex" named options in views.appview.AppView constructor 
 * Added DISQUS in plugins
 * Removed StaticPagesMiddleware request handler
 * response method in djpcmsview class has been replaced with __call__ method
 * Change in urls
 * Added DeploySite model
 * Added Deploy plugin
 * Added jquery.cicle in media. jQuery plugin to handle rotating pictures.
 * Added plugin's url for handling dynamic plugins not connected to a model.
 * Added Contact form plugin.
 * ADDED NEW MODEL AdditionalPageData for injecting ad-hoc data into page head or javascript in page body
 * Content text plugin is now wrapped into a div with class 'djpcms-text-content'.
 * Breadcrumbs name is given by view title
 * Created the DjpResponse object in views.response.

djpcms/settings.py
import os,sys

from django.conf import settings
from djpcms.ajaxhtml import ajaxhtml

# Not available yet. For future releases
LANGUAGE_REDIRECT      = getattr(settings, 'LANGUAGE_REDIRECT', False)
DEFAULT_TEMPLATE_NAME  = getattr(settings, 'DEFAULT_TEMPLATE_NAME', ['base.html','djpcms/base.html'])
DEFAULT_VIEW_MODULE    = getattr(settings, 'DEFAULT_VIEW_MODULE', 'djpcms.views.pageview.pageview')
USE_TINYMCE            = getattr(settings, 'USE_TINYMCE', False)
HTML_CLASSES           = getattr(settings, 'HTML_CLASSES', ajaxhtml())
MAX_SEARCH_DISPLAY     = getattr(settings, 'MAX_SEARCH_DISPLAY', 20)
DEFAULT_CODE_BASKET    = getattr(settings, 'DEFAULT_CODE_BASKET', ('code','user'))
CACHE_VIEW_OBJECTS     = getattr(settings, 'CACHE_VIEW_OBJECTS', True)

DJPCMS_IMAGE_UPLOAD_FUNCTION = getattr(settings, 'DJPCMS_IMAGE_UPLOAD_FUNCTION', None)

SERVE_STATIC_FILES     = getattr(settings, 'SERVE_STATIC_FILES', False)

# Root page for user account urls
USER_ACCOUNT_HOME_URL    = getattr(settings, 'USER_ACCOUNT_HOME_URL', '/accounts/')


JS_START_END_PAGE        = getattr(settings, 'JS_START_END_PAGE', 101)

EXTRA_CONTENT_PLUGIN     = getattr(settings, 'EXTRA_CONTENT_PLUGIN', None)
GOOGLE_ANALYTICS_ID      = getattr(settings, 'GOOGLE_ANALYTICS_ID', None)

ENABLE_BREADCRUMBS       = getattr(settings, 'ENABLE_BREADCRUMBS', 1)

APPLICATION_URL_MODULE   = getattr(settings, 'APPLICATION_URL_MODULE', None)
APPLICATION_URL_PREFIX   = getattr(settings, 'APPLICATION_URL_PREFIX', '/apps/')

DJPCMS_PLUGINS           = getattr(settings, 'DJPCMS_PLUGINS', ['djpcms.plugins.*'])
DJPCMS_WRAPPERS          = getattr(settings, 'DJPCMS_WRAPPERS', ['djpcms.plugins.extrawrappers'])
DJPCMS_CONTENT_FUNCTION  = getattr(settings, 'DJPCMS_CONTENT_FUNCTION', None)
DJPCMS_MARKUP_MODULE     = getattr(settings, 'DJPCMS_MARKUP_MODULE', 'djpcms.utils.markups')

DJPCMS_PLUGIN_BASE_URL   = getattr(settings, 'DJPCMS_PLUGIN_BASE_URL', '/plugin/')



#---------------------------------- Styling

# Default grid
GRID960_DEFAULT_FIXED    = getattr(settings, 'GRID960_DEFAULT_FIXED', True)
# Style css name
DJPCMS_SITE_STYLE        = getattr(settings, 'DJPCMS_SITE_STYLE', 'smoothness')

# Inline editing configuration
# By default it is switch on
#
#    available:    boolean   whether the inline editing is on or off
#    preurl:       slug      which will be used as initial part of the editing url
#                            So if a page has the url /some/path/to/page/ its editing url will
#                            be /preurl/some/path/to/page/
#    permission:   String    dotted path to a function handling editing permissions
#    pagecontent:  String    Code of page to be used as root for site content. This page must be
#                            available in the database.
CONTENT_INLINE_EDITING = getattr(settings, 'CONTENT_INLINE_EDITING',  {'available':True,
                                                                       'preurl': 'edit-content',
                                                                       'permission': None,
                                                                       'pagecontent': '/site-content/',
                                                                       'width': 600,
                                                                       'height': 400})

libpath = os.path.join(os.path.dirname(__file__),'libs')
if libpath not in sys.path:
    sys.path.append(libpath)